src/runtime/runtime_stm32f103.go

```diff
@@ -191,19 +183,10 @@ func timerSleep(ticks uint32) bool {
 	// Enable the timer.
 	stm32.TIM3.CR1.SetBits(stm32.TIM_CR1_CEN)
 
-wait:
-	arm.Asm("wfi")
-	if timerWakeup.Get() != 0 {
-		return true
-	}
-
-	if hasScheduler {
-		return false
-	} else {
-		// keep looping until the routine exits or is interrupted
-		goto wait
+	// Wait for interrupt to fire
+	for timerWakeup.Get() == 0 {
+		arm.Asm("wfi")
```

Comment:
This change means that waking up a goroutine from an interrupt won't work anymore. That's still better than if the chip is non-functional entirely so might be intentional.